### PR TITLE
MINOR: Various cleanups in client tests

### PR DESCRIPTION
**Pull request by mimaison**
- Simplify assertions
- Remove redundant types
- Use lambdas instead of anonymous classes
- Remove unnecessary throws

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java [hunk 1]
```diff
@@ -320,7 +315,7 @@ public void testDoubleBuild() {
         try {
             builder.build();
             fail("Expected calling build twice to fail.");
-        } catch (Throwable t) {
+        } catch (NullPointerException npe) {
```
Comment (divijvaidya) [inline clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java]: NPE is a weird way of enforcing idempotency for build()! While you are in this code base could we add a javadoc to build() to clarify this expectation?

**Review by divijvaidya — APPROVED** [review]
Folks usually do [spring cleaning ](https://en.wikipedia.org/wiki/Spring_cleaning) during spring but you have done it in winters :D 

Thank you for the change. Looks good.

File: clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java (hunk 1 above)
Comment (mimaison) [reply clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java]: You're right! I've pushed an update with a few changes to improve this.

Comment (divijvaidya) [reply clients/src/test/java/org/apache/kafka/clients/FetchSessionHandlerTest.java]: Thank you. Looks good to merge!

File: clients/src/test/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerSaslClientCallbackHandlerTest.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 import org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler;
 import org.junit.jupiter.api.Test;
 
-public class OAuthBearerSaslClienCallbackHandlerTest {
+public class OAuthBearerSaslClientCallbackHandlerTest {
```
Comment (showuon) [inline clients/src/test/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerSaslClientCallbackHandlerTest.java:35]: Aha, nice catch!

File: clients/src/test/java/org/apache/kafka/common/utils/ExitTest.java [hunk 3]
```diff
@@ -34,7 +34,7 @@ public void shouldHaltImmediately() {
         });
         try {
             int statusCode = 0;
-            String message = "mesaage";
+            String message = "message";
```
Comment (showuon) [inline clients/src/test/java/org/apache/kafka/common/utils/ExitTest.java:37]: Wow, how could you find it! Cool!

**Review by showuon — APPROVED** [review]
LGTM! Thanks for the cleanups!

File: clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java [hunk 4]
```diff
@@ -266,7 +266,14 @@ public void add(TopicPartition topicPartition, PartitionData data) {
             }
         }
 
+        /**
+         * Build a FetchRequestData for the provided partitions
+         * @throws IllegalStateException if it has already been called
+         */
         public FetchRequestData build() {
+            if (next == null) {
+                throw new IllegalStateException("build() has already been called.");
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: This is not a test change?

Comment (mimaison) [reply clients/src/main/java/org/apache/kafka/clients/FetchSessionHandler.java]: I've reverted that change, I'll address this issue in another PR.

File: clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java [hunk 5]
```diff
@@ -331,17 +331,15 @@ public void testStressfulSituation() throws Exception {
...
-            threads.add(new Thread() {
-                public void run() {
-                    for (int i = 0; i < msgs; i++) {
-                        try {
-                            accum.append(topic, i % numParts, 0L, key, value, Record.EMPTY_HEADERS, null, maxBlockTimeMs, false, time.milliseconds(), cluster);
-                        } catch (Exception e) {
-                            e.printStackTrace();
-                        }
+            threads.add(new Thread(() -> {
+                for (int i1 = 0; i1 < msgs; i1++) {
```
Comment (clolov) [inline clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java]: Nit: Does this need to be `i1` - are we iterating through the same types of things as `i` is iterating through? Can we leave it as `i` similar to the previous version or change it to `j` (or something similar)?

**Review by clolov — APPROVED** [review]

File: clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java (hunk 5 above)
Comment (mimaison) [reply clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java]: Good spot! I had not realized IntelliJ had automatically named it `i1` which is a bit strange. I renamed it to `j`.

**Review by clolov — APPROVED** [review]
Looks great to me!

**Pull request merged**